### PR TITLE
calc: save edited cell on exit

### PR DESCRIPTION
**Pull request by eszkadev**
This fixes the main problem which is data loss when we
exit without confirmation of a cell content but also
problem with losing focus from formulabar on some actions.
Especially when we click outside iframe - we lost
focus on formulabar but we were still catching input to
it on the server side.

### CODE REVIEW

**Comment by eszkadev** [comment]
I need to fix tests. Focus should be handled somewhere else.

**Comment by eszkadev** [comment]
Now it seems to be fixed. I solved problem with losing focus what was causing issues with detecting when we have editing active...

**Comment by mcecchetti** [comment]
Question: has formula bar focus when a cell is edited in place ?
I'm asking since the patch takes care of the formula bar focus status, only.

**Comment by eszkadev** [comment]
> Question: has formula bar focus when a cell is edited in place ? I'm asking since the patch takes care of the formula bar focus status, only.

@mcecchetti thanks, added also this case

File: browser/src/control/Control.FormulaBarJSDialog.js [hunk 1]
```diff
@@ -283,6 +283,13 @@ class FormulaBar {
 		return L.DomUtil.hasClass(input, 'focused');
 	}
 
+	isInEditMode() {
+		var acceptButton = this.parentContainer.querySelector('#acceptformula');
+		if (acceptButton)
+			return !acceptButton.classList.contains('hidden');
```
Comment (mcecchetti) [inline browser/src/control/Control.FormulaBarJSDialog.js:289]: `return true` ?

**Review by mcecchetti — APPROVED** [review]
Okay, seems to work.
Maybe if someone switch to another window/app and then back it could be a bit unexpected that on typing instead of appending new text, it overwrites the current content.

**Pull request merged**